### PR TITLE
Add Protective Marking sub-section to api-security.md

### PR DESCRIPTION
**Pull request by TimGoodwill**
Addition of Protective Marking sub-section to API Security doc.
A unified approach to protective marking would greatly enhance API inter-operabilty. It is acknowledged that each jurisdiction may have similar requirements, and this section should be open to evolution.

### CODE REVIEW

**Review by cj7hawk — APPROVED** [review]
Will require additional updates by states as state schemes enter API requirements as this section expands in the future.

File: sections/api-security.md [hunk 1]
```diff
@@ -33,11 +35,11 @@ Depending on the security classification you may be required to establish the fo
...
+- Always set a reasonable expiration date for tokens. An OIDC access token lifetime **SHOULD NOT*** exceed 5 minutes.
 - JWT refresh tokens **SHOULD** be used when new JWT tokens are required outside of this lifetime. 
-- All APIs **MUST** have a policy that only allows access based on a valid API key.
-- API keys **MUST** be used for client authentication. Use of API keys should only be permitted when TLS is enabled. Rotation policy for API Key should be implemented as well.
-- API keys **SHOULD NOT** be included in the URL or query string. API keys **SHOULD** be included in the HTTP header as query strings may be saved by the client or server in unencrypted format by the browser or server application. 
+- All APIs **MUST** have a policy that only allows access to an authenticated client application.
+- Client application identity **MUST** be established via a consistent mechanism. This may be via an API Key, or via a more robust mechanism such as an OAuth 2.0 asserted server identity. Use of API keys should only be permitted when TLS is enabled. Rotation policy for API Key/secret should be implemented where applicable.
```
Comment (cj7hawk) [inline sections/api-security.md]: This goes against principles that suggest the mechanism only needs to be strong enough to ensure the relevant security of the client and of the agency moving any data. If the data is not classified, then it seems unreasonable to require an API key for every application. Open APIs for public data for which there would be no restrictions on who can get an API key, or who could see the information should not require an API key to be provided, as this becomes somewhat invasive.  Would recommend changing this to MUST only when there is an information classification level above unclassified.  Does anyone have any other thoughts about this?

**Review by cj7hawk — CHANGES_REQUESTED** [review]
Have made a note where a requirement does not seem to fit requirements. An API key should not be a requirement where no sensitive information is being exchanged.

File: sections/api-security.md (hunk 1 above)
Comment (TimGoodwill) [reply sections/api-security.md]: I don't disagree - I have reworded from the perspective of API Key being an unacceptably weak mechanism for client authentication, but suitable for client identification. If SLAs and the collection of metrics are to apply to public APIs, a client identification mechanism is required, so I would suggest that a SHOULD will apply at the very least. I will reword.

Comment (cj7hawk) [reply sections/api-security.md]: SHOULD would be OK.

**Comment by cj7hawk** [comment]
As per Technical Meeting, approved.

**Pull request merged**